doc/source/whatsnew/v2.0.0.rst

```diff
@@ -77,6 +77,13 @@ be set to ``"pyarrow"`` to return pyarrow-backed, nullable :class:`ArrowDtype` (
         df_pyarrow = pd.read_csv(data, use_nullable_dtypes=True, engine="pyarrow")
     df_pyarrow.dtypes
 
+Copy on write improvements
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Added a new section like is being added in https://github.com/pandas-dev/pandas/pull/50471. We can consolidate those different blurbs later on in a coherent whatsnew.